src/spawn_blocking.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{
+    concurrency_limit::ConcurrencyLimit, error::Error, make_future_cancellable,
+    ComputeHeavyFutureExecutor,
+};
+
+pub(crate) struct SpawnBlockingExecutor {
+    concurrency_limit: ConcurrencyLimit,
+}
+
+impl SpawnBlockingExecutor {
+    pub(crate) fn new(max_concurrency: Option<usize>) -> Self {
+        let concurrency_limit = ConcurrencyLimit::new(max_concurrency);
+
+        Self { concurrency_limit }
+    }
+}
+
+impl ComputeHeavyFutureExecutor for SpawnBlockingExecutor {
+    async fn execute<F, O>(&self, fut: F) -> Result<O, Error>
+    where
+        F: std::future::Future<Output = O> + Send + 'static,
+        O: Send + 'static,
+    {
+        let _permit = self.concurrency_limit.acquire_permit().await;
+
+        let (wrapped_future, rx) = make_future_cancellable(fut);
+
+        if let Err(err) = tokio::task::spawn_blocking(move || {
+            tokio::runtime::Handle::current().block_on(wrapped_future)
+        })
```

Comment:
does this actually work? I think once a bad future is in the runtime, it's going to potentially tie up a worker

Reply:
Hmm, I think I found that approach from something alice posted somewhere, let me dig it up.

It's currently test e2e with a thread::sleep so it seems to work as expected?
https://github.com/jlizen/compute-heavy-future-executor/blob/main/tests/spawn_blocking_strategy.rs#L36

Reply:
I think it was from here, which isn't particularly authoritative
https://stackoverflow.com/questions/76965631/how-do-i-spawn-possibly-blocking-async-tasks-in-tokio

Going to poke around this more closely with a perfetto trace per our offline discussion, thanks for the callout

Reply:
Ah, alice also discusses it in discord eg https://discord.com/channels/500028886025895936/500336333500448798/1189652644881514586

But that doesn't address it being a potentially blocking future

Reply:
To summarize offline discussion - the behavior is actually pretty unexpected. Calling `Handle::block_on()` from inside `spawn_blocking()` just spawns a task to a regular worker thread for that runtime. Which makes it pretty useless for our purposes.

Which kind of makes sense, how else would tokio know how to execute the futures without a runtime, and it doesn't want to lazily spin up a current thread runtime for every spawn blocking thread. At that point probably you're better off managing your own threadpool full of current thread runtimes (or using a delegated secondary multithreaded runtime).

Anyway, PR coming to rip spawn blocking out and default to secondary tokio runtime in case the tokio feature is enabled.

Reply:
I'm also wondering if we should have a sync input API as well, for the 'definitely this is blocking and not containing futures' segments. 

Since, for those, there is a simpler use case that might just involves `spawn_blocking` along with concurrency control.

Today, library authors can pick between `spawn_blocking()` on them (pretty opinionated, also tokio-specific) or just intermingling them with futures (no caller ability to slice them apart).

Reply:
cut https://github.com/jlizen/compute-heavy-future-executor/issues/8